corehq/tabs/tabclasses.py

```diff
@@ -159,10 +159,27 @@ def _get_tools_items(self):
             'icon': 'icon-tasks fa fa-tasks',
             'show_in_dropdown': True,
         }]
-        if toggles.USER_CONFIGURABLE_REPORTS.enabled(self.couch_user.username):
-            # Only show for **users** with the flag. This flag is also available for domains
-            # but should not be granted by domain, as the feature is too advanced to turn
-            # on for all of a domain's users.
+        from corehq.apps.users.models import Permissions
+        # For SaaS envs, if we want to display the tab only if Feature Flag is
+        # enabled on domain and the role has the permission to edit_ucrs
+        show_on_sass_env = (
+            toggles.USER_CONFIGURABLE_REPORTS.enabled(self.domain)
+            and self.couch_user.has_permission(
+                self.domain,
+                get_permission_name(Permissions.edit_ucrs)
+            )
+            and settings.IS_SAAS_ENVIRONMENT
+        )
+        # For Non SaaS Envs
+        # Only show for **users** with the flag. This flag is also available for domains
```

Comment:
is this comment still relevant now that we have user-level permissions for the domain level feature flag?